src/components/Pama/pama.jsx

```diff
@@ -0,0 +1,187 @@
+  },
+  onMessage: ({ data, dispatch }) => {
+    const updates = [data].filter(({ messageType }) => messageType === 'scratchpad.update');
+    dispatchUpdates(dispatch, updates);
+  },
+  generateContext: state => ({
+    draftOrders: {
+      resourceType: 'Bundle',
+      entry: [
+        {
+          resource: {
+            resourceType: 'ServiceRequest',
+            id: 'example-request-id',
+            status: 'draft',
+            intent: 'plan',
+            code: {
+              coding: [
+                {
+                  system: 'http://www.ama-assn.org/go/cpt',
+                  code: state.pama.serviceRequest.code,
+                },
+              ],
+              text: 'Lumbar spine CT',
+            },
+            subject: {
+              reference: `Patient/${state.patientState.currentPatient.id}`,
+            },
+            reasonCode: [
+              {
+                coding: [
+                  {
+                    system: 'http://snomed.info/sct',
+                    code: state.pama.serviceRequest.reasonCode,
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      ],
+    },
+  }),
+};
+
+cdsExecution.registerTriggerHandler('pama/order-select', pamaTriggerHandler);
+
+cdsExecution.registerTriggerHandler('pama/order-sign', {
+  ...pamaTriggerHandler,
+  needExplicitTrigger: 'TRIGGER_ORDER_SIGN',
+});
+
+const studyCodes = [
+  {
+    display: 'Lumbar spine CT',
+    code: '72133',
+  },
+  {
+    display: 'Cardiac MRI',
+    code: '75561',
+  },
+];
+
+const reasonCodes = [
+  {
+    display: 'Low back pain',
+    code: '279039007',
+  },
+  {
+    display: 'Congenital heart disease',
+    code: '13213009',
+  },
+];
+
+export class OrderImaging extends Component {
+  updateField(field) {
+    return (e, v) => this.props.updateServiceRequest(field, v);
+  }
+
+  render() {
+    const { pamaRating } = this.props;
+    const { code, reasonCode } = this.props.serviceRequest;
+
+    return (
+      <div>
+        <Select
+          name="study-select"
+          value={code}
+          onChange={this.updateField('code')}
+        >
+          {studyCodes.map(coding => (
+            <Select.Option
+              key={coding.code}
+              value={coding.code}
+              display={coding.display}
+            />
+          ))}
+        </Select>
+        <Select
+          name="reason-select"
+          value={reasonCode}
+          onChange={this.updateField('reasonCode')}
+        >
+          {reasonCodes.map(coding => (
+            <Select.Option
+              key={coding.code}
+              value={coding.code}
+              display={coding.display}
+            />
+          ))}
+        </Select>
+
+        <span>
+          PAMA Rating: {pamaRating}
+          {{ appropriate: '✓', 'not-appropriate': '⚠' }[pamaRating] || '?'}
+        </span>
+        <br />
+        <Button text="Sign Order" onClick={this.props.signOrder} />
```

Comment:
How does this work (or intended to work) with the CDS Developer Panel?

Reply:
It's a bit of an open UX question -- I'm thinking the CDS Developer Panel would maybe include all hooks defined on the current screen, and would show the most recently called service? Still TBD, so I'm removing this button for now. (I just wanted to make sure we'd thought through how to *trigger* an explicit hook like `order-sign`.)